src/core/server/metrics/collectors/process.test.ts

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import v8, { HeapInfo } from 'v8';
+import { ProcessMetricsCollector } from './process';
+
+describe('ProcessMetricsCollector', () => {
+  let collector: ProcessMetricsCollector;
+
+  beforeEach(() => {
+    collector = new ProcessMetricsCollector();
+  });
+
+  afterEach(() => {
+    jest.restoreAllMocks();
+  });
+
+  it('collects pid from the process', async () => {
+    const metrics = await collector.collect();
+
+    expect(metrics.pid).toEqual(process.pid);
+  });
+
+  it('collects event loop delay', async () => {
+    const metrics = await collector.collect();
+
+    expect(metrics.event_loop_delay).toBeGreaterThan(0);
+  });
```

Comment:
Mocking `getEventLoopDelay` is a pain, so I only tested that we collect an actual > 0 value here.